File: frontend/demo/component/messages/message-list-one-to-one.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/message-list';
import { format, subMinutes } from 'date-fns';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import type { MessageListItem } from '@vaadin/message-list';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('message-list-one-to-one')
export class Example extends LitElement {
  private isoMinutes = 'yyyy-MM-dd HH:mm';
  private fiveMinutesAgo = format(subMinutes(new Date(), 5), this.isoMinutes);
  private fourMinutesAgo = format(subMinutes(new Date(), 4), this.isoMinutes);

  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  // tag::snippet[]
  private items: MessageListItem[] = [
    {
      text: 'Which orders are still waiting for shipment?',
      time: this.fiveMinutesAgo,
      userName: 'Linsey Listy',
      theme: 'self',
    },
    {
      text: `These orders are waiting for shipment:

- **#1042**, paid yesterday
- **#1043**, paid today
- **#1045**, waiting for stock

The first two can ship today. Order #1045 ships when the missing items arrive.`,
      time: this.fourMinutesAgo,
      userName: 'Assistant',
      theme: 'full-width',
    },
  ];

  protected override render() {
    return html`
      <vaadin-message-list
        theme="bubble one-to-one"
        .items="${this.items}"
        markdown
      ></vaadin-message-list>
    `;
  }
  // end::snippet[]
}

File: src/main/java/com/vaadin/demo/component/messages/MessageListOneToOne.java
package com.vaadin.demo.component.messages;

import java.time.Instant;
import java.time.temporal.ChronoUnit;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.messages.MessageList;
import com.vaadin.flow.component.messages.MessageListItem;
import com.vaadin.flow.component.messages.MessageListItemVariant;
import com.vaadin.flow.component.messages.MessageListVariant;
import com.vaadin.flow.router.Route;

@Route("message-list-one-to-one")
public class MessageListOneToOne extends Div {

    public MessageListOneToOne() {
        Instant fiveMinsAgo = Instant.now().minus(5, ChronoUnit.MINUTES);
        Instant fourMinsAgo = Instant.now().minus(4, ChronoUnit.MINUTES);

        // tag::snippet[]
        MessageList list = new MessageList();
        list.addThemeVariants(MessageListVariant.BUBBLE,
                MessageListVariant.ONE_TO_ONE);
        list.setMarkdown(true);

        MessageListItem prompt = new MessageListItem(
                "Which orders are still waiting for shipment?", fiveMinsAgo,
                "Linsey Listy");
        prompt.addThemeVariants(MessageListItemVariant.SELF);

        MessageListItem response = new MessageListItem(
                """
                        These orders are waiting for shipment:

                        - **#1042**, paid yesterday
                        - **#1043**, paid today
                        - **#1045**, waiting for stock

                        The first two can ship today. Order #1045 ships when the missing items arrive.
                        """,
                fourMinsAgo, "Assistant");
        response.addThemeVariants(MessageListItemVariant.FULL_WIDTH);

        list.setItems(prompt, response);
        add(list);
        // end::snippet[]
    }

    public static class Exporter extends DemoExporter<MessageListOneToOne> { // hidden-source-line
    } // hidden-source-line
}
